pykg2tbl/service.py

```diff
@@ -102,55 +169,55 @@ class KG2EndpointSource(KGSource):
     :param url: url of the endpoint to make the KGSource from.
     """
 
-    def __init__(self, url):
+    def __init__(self, *url):
         super().__init__()
-        self.endpoint = url
+        self.endpoints = [f for f in url]
 
-    # TODO decide later on proper conversion to remove rdflib specifics and
-    #   create reusable data dict for conversion through query results
-    #   (pandas wrapper)
     @staticmethod
-    def reslist_to_dict(reslist: list):
+    def query_result_to_dict(reslist: list):
         return [
             {k: row[k]["value"] for k in row}
             for row in reslist["results"]["bindings"]
         ]
 
     def query(self, sparql: str) -> QueryResult:
-        ep = SPARQLWrapper(self.endpoint)
-        ep.setQuery(sparql)
-        ep.setReturnFormat("json")
-        reslist = ep.query().convert()
-        return QueryResult(KG2EndpointSource.reslist_to_dict(reslist))
+        reslist = []
+        for url in self.endpoints:
+            ep = SPARQLWrapper(url)
+            ep.setQuery(sparql)
+            ep.setReturnFormat("json")
+            resdict = ep.query().convert()
+            reslist = reslist + self.query_result_to_dict(resdict)
 
+        query_result = QueryResult(reslist)
+        return query_result
 
-class SparqlBuilder(ABC):
-    @abstractmethod
-    def build_sparql_query(self, name: str, **variables):
-        """
-        Builds the named sparql query by applying the provided params
 
-        :param name: Name of the query.
-        :param variables: Dict of all the variables given to the template to
-            make the sparql query.
+def check_source(source: Union[str, Tuple[str, ...], List]) -> str:
+    if isinstance(source, tuple) or isinstance(source, list):
+        return check_source(source[0])
+    source_type = "file"
+    if source.startswith("http"):
+        source_type = "endpoint"
```

Comment:
@JotaFan this will  not be enough
the url passed might be pointing to a json-ld or turtle file that would effectively have the remote-file-content for a File-Source -- and not a sparql end-point !

so in case of starts-with http you will need some more testing - possible approach might be:
* check the content-type return 
*  find some test one could use on a sparql endpoint ?
  * possibly use this list of [example endpoints](https://www.w3.org/wiki/SparqlEndpoints) to verify your logic
  * do a spo query with limit 1
  * or check the spec for some description service all endpoints should support ?

Reply:
Yes!
Thanks, I should have written this down before.
I did  not do it yet because I wanted to get the classes cleaned first, but I was already forgetting about this.
I'm on it now